user_guide_src/source/dbmgmt/forge.rst

```diff
@@ -379,123 +379,135 @@ change the name, you can add a "name" key into the field defining array.
 
 ::
 
-	$fields = [
-		'old_name' => [
-			'name' => 'new_name',
-			'type' => 'TEXT',
-		],
-	];
-	$forge->modifyColumn('table_name', $fields);
-	// gives ALTER TABLE table_name CHANGE old_name new_name TEXT
+    $fields = [
+        'old_name' => [
+            'name' => 'new_name',
+            'type' => 'TEXT',
+        ],
+    ];
+    $forge->modifyColumn('table_name', $fields);
+    // gives ALTER TABLE table_name CHANGE old_name new_name TEXT
 
 ***************
 Class Reference
 ***************
 
 .. php:class:: CodeIgniter\\Database\\Forge
 
-	.. php:method:: addColumn($table[, $field = []])
+    .. php:method:: addColumn($table[, $field = []])
 
-		:param	string	$table: Table name to add the column to
-		:param	array	$field: Column definition(s)
-		:returns:	true on success, false on failure
-		:rtype:	bool
+        :param    string    $table: Table name to add the column to
+        :param    array    $field: Column definition(s)
+        :returns:    true on success, false on failure
+        :rtype:    bool
 
-		Adds a column to a table. Usage:  See `Adding a Column to a Table`_.
+        Adds a column to a table. Usage:  See `Adding a Column to a Table`_.
 
-	.. php:method:: addField($field)
+    .. php:method:: addField($field)
 
-		:param	array	$field: Field definition to add
-		:returns:	\CodeIgniter\Database\Forge instance (method chaining)
-		:rtype:	\CodeIgniter\Database\Forge
+        :param    array    $field: Field definition to add
+        :returns:    \CodeIgniter\Database\Forge instance (method chaining)
+        :rtype:    \CodeIgniter\Database\Forge
 
                 Adds a field to the set that will be used to create a table. Usage:  See `Adding fields`_.
 
-	.. php:method:: addKey($key[, $primary = false[, $unique = false]])
+    .. php:method:: addForeignKey($fieldName, $tableName, $tableField[, $onUpdate = '', $onDelete = ''])
```

Comment:
Is this okay?

In fact, the method signature is:
`addForeignKey($fieldName = '', string $tableName = '', $tableField = '', string $onUpdate = '', string $onDelete = '')`

But it seems `$fieldName`, `$tableName`, `$tableField` are required.
Why do they have the default value `''`?

Reply:
It seems to me a design fail.